dask/array/tests/test_linalg.py

```diff
@@ -97,6 +97,23 @@ def test_linalg_consistent_names():
     assert same_keys(v1, v2)
 
 
+@pytest.mark.parametrize("m,n", [
+    (10, 20),
+    (15, 15),
+    (20, 10),
+])
+def test_dask_svd_self_consistent(m, n):
+    a = np.random.rand(m, n)
+    d_a = da.from_array(a, chunks=(3, n), name='A')
+
+    d_u, d_s, d_vt = da.linalg.svd(d_a)
+    u, s, vt = da.compute(d_u, d_s, d_vt)
+
+    for d_e, e in zip([d_u, d_s, d_vt], [u, s, vt]):
+        assert d_e.shape == e.shape
+        assert d_e.dtype == e.dtype
```

Comment:
Can we check for these by using `assert_eq`?

Reply:
We could, but that would be overkill as we already know the results are exactly equal.

Reply:
So there is also an issue between NumPy and Dask implementations as well, but let's save that discussion for issue ( https://github.com/dask/dask/issues/3576 ). While we figure that out, it would be good if the Dask Arrays at least didn't misreport the data they hold.

Had looked into including this in another test initially, but the other tests seem to have some pretty niche concerns, which made it unnatural to add this to them and convoluted the otherwise simple logic included here. Plus we want to test several possible shapes, which may have slowed down tests that were comparing results. Given this, opted to breakout the test instead.

Since this test is pretty short, clean, and fast, not sure it is worth spending time trying to stuff it into another test that is running cross-purpose to this one.